source/AlertLabel.cpp

```diff
@@ -0,0 +1,94 @@
+/* AlertLabel.cpp
+Copyright (c) 2022 by Daniel Yoon
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "AlertLabel.h"
+
+#include "Angle.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "Government.h"
+#include "LineShader.h"
+#include "Projectile.h"
+#include "pi.h"
+#include "RingShader.h"
+#include "Ship.h"
+#include "Sprite.h"
+
+#include <algorithm>
+#include <cmath>
+
+using namespace std;
+
+namespace {
+	const double LINE_ANGLE[4] = {60., 120., 300., 240.};
+	const double LINE_LENGTH = 6.;
+	const double INNER_SPACE = 1.;
+	const double LINE_GAP = 1.;
+	const double GAP = 4.;
+	const double MIN_DISTANCE = 1.;
+}
+
+
+
+AlertLabel::AlertLabel(Point &position, const Projectile &projectile, shared_ptr<Ship> flagship, double zoom) : position(position * zoom)
+{
+
+	const bool isTargetingMe = projectile.TargetPtr() == flagship;
+	const Government *gov = projectile.GetGovernment();
+	const bool isInDanger = projectile.GetWeapon().HullDamage() > flagship->Attributes().Get("hull")/10;
```

Comment:
how about instead of having two states, interpolate between two states

Reply:
Outdated, but hey are now separate flags that can be active independently.